tests/conftest.py

```diff
@@ -46,32 +48,70 @@ def start_mock_server():
     port = utils.free_port()
     root = os.path.abspath(os.path.join(os.path.dirname(__file__), ".."))
     path = os.path.join(root, "tests", "utils", "mock_server.py")
-    command = ["python", path]
+    command = [sys.executable, "-u", path]
     env = os.environ
     env["PORT"] = str(port)
     env["PYTHONPATH"] = root
-    server = subprocess.Popen(command, env=env)
+    logfile = open("tests/logs/live_mock_server.log", "w")
+    server = subprocess.Popen(command, stdout=logfile,
+                              env=env, stderr=subprocess.STDOUT, bufsize=1,
+                              close_fds=True)
     server._port = port
     server.base_url = "http://localhost:%i" % server._port
+
+    def get_ctx():
+        return requests.get(server.base_url + "/ctx").json()
+
+    def set_ctx(payload):
+        return requests.put(server.base_url + "/ctx", json=payload).json()
+
+    def reset_ctx():
+        return requests.delete(server.base_url + "/ctx").json()
+
+    server.get_ctx = get_ctx
+    server.set_ctx = set_ctx
+    server.reset_ctx = reset_ctx
```

Comment:
This is hot, `live_mock_server` now let's you query for the state of the server to make assertions in tests.

Reply:
wild